src/views/PageAssembler.vue

```diff
@@ -122,19 +121,23 @@ export default {
 		},
 
 		isItemShown (item) {
-			if (this.getter(item.hideDependOnGetter)?.error)
+			if (this.getter(item.hideDependOnGetter)?.error) {
 				return false;
+			}
 
-			if (item.hideEmptyData && (
-				!this.getData(item) || (
-					Array.isArray(this.getData(item)) && !this.getData(item)?.length
-				)
-			)
-			)
-				return false;
+			if (item.hideEmptyData) {
+				const data = this.getData(item);
+
+				if (Array.isArray(data)) {
+					return !!data.length;
+				} else {
+					return !!Object.keys(data).length
```

Comment:
I guess this should work also for arrays, so nested `if` statement is not necessary here.

Reply:
Also I would suggest to change this part only to return `false` when `hideEmptyData` is true and  `Object.keys(data).length` is 0.
Otherwise when `hideEmptyData` is positive, we will never reach the part below (and all errored views will be rendered):
`[L138] if (item.hideOnError && this.getter(item.managerGetter)?.error)`

Reply:
> Also I would suggest to change this part only to return `false` when `hideEmptyData` is true and `Object.keys(data).length` is 0.
> Otherwise when `hideEmptyData` is positive, we will never reach the part below (and all errored views will be rendered):
> `[L138] if (item.hideOnError && this.getter(item.managerGetter)?.error)`

that makes sense 👍🏼